### ISSUE TITLE
Unable to add external dependencies for package

### ISSUE THREAD

**Issue by jprosevear** [TanStack/config#42]
### Describe the bug

If I add external deps in my vite.config.ts and merge with the tanstack build, my external deps are lost and only the tanstack/config deps generated by 'vite-plugin-externalize-deps' are available.

### Steps to reproduce

```
import { tanstackBuildConfig } from '@tanstack/config/build';
import { defineConfig, mergeConfig } from 'vitest/config';
import react from '@vitejs/plugin-react';

const config = defineConfig({
  plugins: [react()],
  build: {
    rollupOptions: {
      external: ['rhino.config', 'virtual:@rhino-project/config/env']
    }
  }
});

export default mergeConfig(
  config,
  tanstackBuildConfig({
    entry: ['./src/index.ts', './src/env.ts'],
    srcDir: './src'
  })
);
```

### Expected behavior

Have a method to preserve externals.  In my case I am trying to externalize a virtual module so I can't use peerDeps, devDeps, etc to achieve this.

### How often does this bug happen?

None

### TanStack Config version

0.4.4

### TypeScript version

5.3.2

**Comment by jprosevear** [comment]
I'd like to send a patch that expands the config options to include something like 

Currently my work around is:
```
const config = defineConfig({
  plugins: [
    externalizeDeps({
      include: ['rhino.config', 'virtual:@rhino-project/config/env']
    }),
    react()
  ]
});
```

**Comment by jprosevear** [comment]
Is it reasonable if I extend the options:

```
export type Options = {
  /** Entry file, e.g. `./src/index.ts` */
  entry: string | string[]
  /** Source directory used for type generation, e.g. `./src` */
  srcDir: string
  /** Excluded from type generation, e.g. `[./src/tests]` */
  exclude?: string[]
  /** Excluded from type generation, e.g. `[./src/tests]` */
  external?: string[]
}
```

**Comment by lachlancollins** [comment]
Added in #43

**Comment by lachlancollins** [comment]
@jprosevear I've fixed the behaviour in #47 , but I'm not convinced it works as expected - can you please test for your use case?
